service/entityresolution/claims/claims_entity_resolution.go

```diff
@@ -0,0 +1,141 @@
+	_ *logger.Logger,
+) (entityresolution.CreateEntityChainFromJwtResponse, error) {
+	entityChains := []*authorization.EntityChain{}
+	// for each token in the tokens form an entity chain
+	for _, tok := range req.GetTokens() {
+		entities, err := getEntitiesFromToken(tok.GetJwt())
+		if err != nil {
+			return entityresolution.CreateEntityChainFromJwtResponse{}, err
+		}
+		entityChains = append(entityChains, &authorization.EntityChain{Id: tok.GetId(), Entities: entities})
+	}
+
+	return entityresolution.CreateEntityChainFromJwtResponse{EntityChains: entityChains}, nil
+}
+
+func EntityResolution(_ context.Context,
+	req *entityresolution.ResolveEntitiesRequest, logger *logger.Logger,
+) (entityresolution.ResolveEntitiesResponse, error) {
+	payload := req.GetEntities()
+	var resolvedEntities []*entityresolution.EntityRepresentation
+
+	for idx, ident := range payload {
+		var entityStruct = &structpb.Struct{}
+		switch ident.GetEntityType().(type) {
+		case *authorization.Entity_Claims:
+			claims := ident.GetClaims()
+			if claims != nil {
+				err := claims.UnmarshalTo(entityStruct)
+				if err != nil {
+					return entityresolution.ResolveEntitiesResponse{}, fmt.Errorf("error unpacking anypb.Any to structpb.Struct: %w", err)
+				}
+			}
+		default:
+			retrievedStruct, err := entityToStructPb(ident)
+			if err != nil {
+				logger.Error("unable to make entity struct", slog.String("error", err.Error()))
+				return entityresolution.ResolveEntitiesResponse{}, fmt.Errorf("unable to make entity struct: %w", err)
+			}
+			entityStruct = retrievedStruct
+		}
+		// make sure the id field is populated
+		originialID := ident.GetId()
+		if originialID == "" {
+			originialID = auth.EntityIDPrefix + fmt.Sprint(idx)
+		}
+		resolvedEntities = append(
+			resolvedEntities,
+			&entityresolution.EntityRepresentation{
+				OriginalId:      originialID,
+				AdditionalProps: []*structpb.Struct{entityStruct},
+			},
+		)
+	}
+	return entityresolution.ResolveEntitiesResponse{EntityRepresentations: resolvedEntities}, nil
+}
+
+func getEntitiesFromToken(jwtString string) ([]*authorization.Entity, error) {
+	token, err := jwt.ParseString(jwtString, jwt.WithVerify(false), jwt.WithValidate(false))
+	if err != nil {
+		return nil, fmt.Errorf("error parsing jwt: %w", err)
+	}
+
+	claims := token.PrivateClaims()
+	entities := []*authorization.Entity{}
+
+	// Convert map[string]interface{} to *structpb.Struct
+	structClaims, err := structpb.NewStruct(claims)
+	if err != nil {
+		return nil, fmt.Errorf("error converting to structpb.Struct: %w", err)
+	}
+
+	// Wrap the struct in an *anypb.Any message
+	anyClaims, err := anypb.New(structClaims)
+	if err != nil {
+		return nil, fmt.Errorf("error wrapping in anypb.Any: %w", err)
+	}
+
+	entities = append(entities, &authorization.Entity{
+		EntityType: &authorization.Entity_Claims{Claims: anyClaims},
+		Id:         "jwtentity-claims",
+		Category:   authorization.Entity_CATEGORY_SUBJECT,
```

Comment:
entity in chain is a subject, should we also include an environmental entity? (similar to how we do client id in keycloak ers?)